spec/reek/smells/too_many_instance_variables_spec.rb

```diff
@@ -1,32 +1,79 @@
 require_relative '../../spec_helper'
 require_lib 'reek/smells/too_many_instance_variables'
-require_relative 'smell_detector_shared'
 
 RSpec.describe Reek::Smells::TooManyInstanceVariables do
-  let(:detector) { build(:smell_detector, smell_type: :TooManyInstanceVariables) }
+  context 'reporting smell' do
+    it 'reports the smelly line' do
+      src = <<-EOS
+        class Empty
+          def ivars
+            @a = @b = @c = @d = 1
+            @e = 1
+          end
+        end
+      EOS
+      expect(src).to reek_of(:TooManyInstanceVariables, lines: [1])
+    end
 
-  it_should_behave_like 'SmellDetector'
+    it 'reports the smell parameters' do
+      src = <<-EOS
+        class Empty
+          def ivars
+            @a = @b = @c = @d = 1
+            @e = 1
+          end
+        end
+      EOS
+      expect(src).to reek_of(:TooManyInstanceVariables, count: 5)
+    end
 
-  def default_max_ivars
-    Reek::Smells::TooManyInstanceVariables::DEFAULT_MAX_IVARS
-  end
+    it 'reports the smelly context' do
+      src = <<-EOS
+        class Empty
+          def ivars
+            @a = @b = @c = @d = 1
+            @e = 1
+          end
+        end
+      EOS
+      expect(src).to reek_of(:TooManyInstanceVariables, context: 'Empty')
+    end
 
-  def too_many_ivars
-    default_max_ivars + 1
-  end
+    it 'reports the smell message' do
+      src = <<-EOS
+        class Empty
+          def ivars
+            @a = @b = @c = @d = 1
+            @e = 1
+          end
+        end
+      EOS
+
+      message = 'has at least 5 instance variables'
 
-  def ivar_sequence(count: default_max_ivars, alphabet: ('a'..'z').to_a)
-    alphabet.first(count).map do |name|
-      "@#{name}=#{rand}"
-    end.join(',')
+      expect(src).to reek_of(:TooManyInstanceVariables, message: message)
```

Comment:
In my case it's more false-negative avoidance than a TDD thing. However I agree it definitely is unnecessary and #1011 rocks.